src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -185,6 +188,95 @@ namespace OpenMS
     return integral;
   }
 
+  PeakIntegrator::PeakShapeMetrics_ PeakIntegrator::calculatePeakShapeMetrics(
+    const MSChromatogram& chromatogram,
+    const double& left,
+    const double& right
+  ) const
+  {
+    // TODO remove the three following lines
+    // peak_height_ = 965356;
+    // peak_apex_rt_ = 2.7045;
+    // background_height_ = 723.5;
+    PeakShapeMetrics_ peakShapeMetrics;
+    peakShapeMetrics.points_across_baseline = 0;
+
+    const double start_intensity = std::max(chromatogram.RTBegin(left)->getIntensity() - background_height_, 0.0);
```

Comment:
I would not subtract the background_height_ or background_area_ anywhere in this code.  The idea of the class is to give the user the option to subtract out the background if they want.  However, no calculation should be done on the peak_height_ or peak_area_ with the background_height_ or background_area_, respectively, subtracted out.  That maybe why the results are different.

Reply:
It was something present already in the other class: https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp#L319